hls-plugin-api/src/Ide/Plugin/Config.hs

```diff
@@ -47,25 +47,19 @@ data CheckParents
 -- will be surprises relating to config options being ignored, initially though.
 data Config =
   Config
-    { checkParents        :: CheckParents
-    , checkProject        :: !Bool
-    , hlintOn             :: !Bool
-    , diagnosticsOnChange :: !Bool
-    , liquidOn            :: !Bool
-    , formatOnImportOn    :: !Bool
-    , formattingProvider  :: !T.Text
-    , maxCompletions      :: !Int
-    , plugins             :: !(Map.Map T.Text PluginConfig)
+    { checkParents       :: CheckParents
+    , checkProject       :: !Bool
+    , hlintOn            :: !Bool
```

Comment:
where is `hlintOn` used? Suspicious

Reply:
The hlint plugin does use it in its `GetHlintDiagnostics` rule

Reply:
Thanks, this one I foolishly assumed that the comment in the doc was all there was - more purging to do.

Reply:
`hlintOn` is in the plugin for backwards compat in the deprecation process but that was time ago so it can be deleted
@michaelpj feel free to remove it here or we can do later